File: ios/ios/Features/StatusCheck/StatusCheckService.swift
import Foundation

struct StatusCheckService {

    private func backendHost() -> String {
        let args = ProcessInfo.processInfo.arguments
        if let i = args.firstIndex(of: "-backendHost"), i + 1 < args.count {
            return args[i + 1]
        }
        return "127.0.0.1"
    }

    func getStatus() async throws -> StatusCheckModel {
        let host = backendHost()
        let url = URL(string: "http://\(host):8080/health")!
        let (data, _) = try await URLSession.shared.data(from: url)
        return try StatusCheckParser.parse(data: data)
    }
}

File: android/app/src/main/java/com/slomako/android/features/StatusCheckService.kt
package com.slomako.android.features

import com.slomako.android.core.network.ApiError
import okhttp3.OkHttpClient
import okhttp3.Request

class StatusCheckService(
    private val client: OkHttpClient = OkHttpClient()
) {

    fun getStatus(host: String): StatusCheckModel {
        val url = "http://$host:8080/health"
        val request = Request.Builder().url(url).get().build()

        try {
            client.newCall(request).execute().use { response ->
                val body = response.body.string()
                if (!response.isSuccessful) throw ApiError.Http(response.code, body)
                return StatusCheckParser.parse(body)
            }
        } catch (e: ApiError) {
            throw e
        } catch (t: Throwable) {
            throw ApiError.Network(t)
        }
    }
}